### DISCUSSION TITLE
League Slime Volleyball Updates

### DISCUSSION THREAD

**Discussion by zxzzz0** [opendilab/DI-engine#61] [General]
Hello DI-engine community,

Good work for the framework. Open a thread to track PR 23.

## Background
Today is Sep 17, 2021. 1.5 months ago, on Aug 2, 2021 [this issue](https://github.com/opendilab/DI-engine/issues/17) was raised up trying to add a new env into `di-zoo`. Initial version of [PR 23](https://github.com/opendilab/DI-engine/pull/23) was created as early as Aug 12, 2021. However there might be some issues happened. Since then, every time people checked this branch they didn't get too much useful information in the past 5 weeks.

## Plan
We truly believe that adding a new env into `di-zoo` is an easy effort thanks to the good design of `di-engine`. In the future, let's post latest bugs/problems here to bring more transparency in this PR and iterate faster.

Best
zxzzz0

<br>
<br>
<br>
<br>
<br>
screenshot
<img src="https://camo.githubusercontent.com/7eba6ff826871a5f1b9fb48d5dc7472dbf6bdbcda80bfb975bdde5a0ff71fdf5/68747470733a2f2f6f746f726f2e6e65742f696d672f736c696d6567796d2f706978656c2e676966" width="30%">

**Comment by zxzzz0** [comment]
@PaParaZz1 Would you take a look at this next week?

**Reply by PaParaZz1** [reply]
We will continue to update this PR23 since 10.1.

**Comment by zxzzz0** [comment]
We were checking the code for doing mutation. However [here](https://github.com/opendilab/DI-engine/blob/main/ding/league/one_vs_one_league.py#L101) it looks like the mutation is a no-op. Generally speaking, in mutation we would like the agent to be reset to one of the following randomly:
(1) A pre-trained supervised model
(2) Past historical model
(3) Pure random model

**Reply by PaParaZz1** [reply]
random uniformly from all the possible model?

**Reply by zxzzz0** [reply]
It will be a smart selection from past models. (2) and (3) are optional, which can be ignored for now.   
But (1) is needed and the code there seems to be no-op. Why?

**Comment by PaParaZz1** [comment]
We will continue to league training in this issue #90.

**Comment by zxzzz0** [comment]
@PaParaZz1 Would you have time to take a look at it this week? #90 Can simply copy and paste the code from league demo.
